lib/pdk/util/puppet_version.rb

```diff
@@ -27,6 +41,25 @@ def latest_available
         latest
       end
 
+      def fetch_puppet_dev
+        unless PDK::Util::Git.remote_repo? puppet_dev_path
+          FileUtils.mkdir_p puppet_dev_path
+          clone_result = PDK::Util::Git.git('clone', DEFAULT_PUPPET_DEV_URL, puppet_dev_path)
+          unless clone_result[:exit_code].zero?
+            PDK.logger.error clone_result[:stdout]
+            PDK.logger.error clone_result[:stderr]
+            raise PDK::CLI::FatalError, _("Unable to clone git repository at '%{repo}'.") % { repo: DEFAULT_PUPPET_DEV_URL }
+          end
```

Comment:
Might be worth returning early after the clone since the subsequent `fetch` is unlikely to bring in any new commits.